virtcontainers/store/manager.go

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package store
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sync"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/sirupsen/logrus"
+)
+
+// Item represents a virtcontainers items that will be managed through the store.
+type Item uint8
+
+const (
+	// Configuration represents a configuration item to be stored
+	Configuration Item = iota
+
+	// State represents a state item to be stored.
+	State
+
+	// Network represents a networking item to be stored.
+	Network
+
+	// Hypervisor represents an hypervisor item to be stored.
+	Hypervisor
+
+	// Agent represents a agent item to be stored.
+	Agent
+
+	// Process represents a container process item to be stored.
+	Process
+
+	// Lock represents a lock item to be stored.
+	Lock
+
+	// Mounts represents a set of mounts related item to be stored.
+	Mounts
+
+	// Devices represents a set of devices related item to be stored.
+	Devices
+
+	// DeviceIDs represents a set of reference IDs item to be stored.
+	DeviceIDs
+)
+
+func (i Item) String() string {
+	switch i {
+	case Configuration:
+		return "Configuration"
+	case State:
+		return "State"
+	case Network:
+		return "Network"
+	case Hypervisor:
+		return "Hypervisor"
+	case Agent:
+		return "Agent"
+	case Process:
+		return "Process"
+	case Lock:
+		return "Lock"
+	case Mounts:
+		return "Mounts"
+	case Devices:
+		return "Devices"
+	case DeviceIDs:
+		return "Device IDs"
+	}
+
+	return ""
+}
+
+// Store is an opaque structure representing a virtcontainers Store.
+type Store struct {
+	sync.RWMutex
+	ctx context.Context
+
+	root   string
```

Comment:
This seems to be the mandatory key for a store. But why not just call it `name`?